app/forms/new_user_base_form.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+##
+# This class is used to validate user data filled in
+# +app/views/users/new.html.haml+ and +app/views/users/add_permissions.html.haml+.
+class NewUserBaseForm < BaseForm
+  # validates +email+ against duplication
+  validate :email_not_duplicated, if: -> { email.present? }
+
+  attr_reader :email_duplicated
+
+  private
+
+  attr_reader :account_id, :name, :email
+
+  # Checks if +email+ are unique.
+  # If not, add error message to +email+.
+  def email_not_duplicated
+    return if email_unique?
+
+    @email_duplicated = true
+    error_message = I18n.t('add_new_user_form.errors.email_duplicated')
+    errors.add(:email, :invalid, message: error_message)
+  end
```

Comment:
can you just call method name in controller?
`if form.email_duplicated?`
```suggestion
  def email_not_duplicated
    return if email_unique?
    errors.add(:email, :invalid, message: I18n.t('add_new_user_form.errors.email_duplicated'))
  end
```